src/core/font_substitutions.js

```diff
@@ -0,0 +1,464 @@
+      fallback: "Helvetica-BoldItalic",
+    },
+  ],
+  [
+    "Calibri",
+    {
+      prepend: ["Calibri", "Carlito"],
+      style: NORMAL,
+      fallback: "Helvetica",
+    },
+  ],
+  [
+    "Calibri-Bold",
+    {
+      local: {
+        alias: "Calibri",
+        append: "Bold",
+      },
+      style: BOLD,
+      fallback: "Helvetica-Bold",
+    },
+  ],
+  [
+    "Calibri-Italic",
+    {
+      local: {
+        alias: "Calibri",
+        append: "Italic",
+      },
+      style: ITALIC,
+      fallback: "Helvetica-Italic",
+    },
+  ],
+  [
+    "Calibri-BoldItalic",
+    {
+      local: {
+        alias: "Calibri",
+        append: "Bold Italic",
+      },
+      style: BOLDITALIC,
+      fallback: "Helvetica-BoldItalic",
+    },
+  ],
+]);
+
+/**
+ * Create the src path to use to load a font (see FontFace).
+ * @param {Array<String>} prepend A list of font names to search first.
+ * @param {Array<String>|Object} local A list of font names to search. If an
+ *   Object is passed, then local.alias is the name of an other substition font
+ *   and local.append is a String to append to the list of fonts in the alias.
+ *   For example if local.alias is "Foo" and local.append is "Bold" then the
+ *   list of fonts will be "FooSubst1 Bold", "FooSubst2 Bold", etc.
+ * @returns an String with the local fonts.
+ */
+function makeLocal(prepend, local) {
+  let append = "";
+  if (!Array.isArray(local)) {
+    // We are getting our list of fonts in the alias and we'll append Bold,
+    // Italic or both.
+    append = ` ${local.append}`;
+    local = substitutionMap.get(local.alias).local;
+  }
+  let prependedPaths = "";
+  if (prepend) {
+    prependedPaths = prepend.map(name => `local(${name})`).join(",") + ",";
+  }
+  return (
+    prependedPaths + local.map(name => `local(${name}${append})`).join(",")
+  );
+}
+
+/**
+ * Get a font substitution for a given font.
+ * The general idea is to have enough information to create a CSS rule like
+ * this:
+ *   @font-face {
+ *    font-family: 'Times';
+ *    src: local('Times New Roman'), local('Subst1'), local('Subst2'),
+ *         url(.../TimesNewRoman.ttf)
+ *    font-weight: normal;
+ *    font-style: normal;
+ *   }
+ * or use the FontFace API.
+ *
+ * @param {Object} idFactory The ids factory.
+ * @param {String} localFontPath Path to the fonts directory.
+ * @param {String} baseFontName The font name to be substituted.
+ * @param {String} standardFontName The standard font name to use if the base
+ *   font is not available.
+ * @returns an Object with the CSS, the loaded name, the src and the style.
+ */
+function getFontSubstitution(
+  idFactory,
+  localFontPath,
+  baseFontName,
+  standardFontName
+) {
+  let mustAddBaseFont = false;
+
+  // It's possible to have a font name with spaces, commas or dashes, hence we
+  // just replace them by a dash.
+  baseFontName = baseFontName
+    .replaceAll(/[\s\-,_]+/g, "-")
+    .replace(/Bold-Italic|Italic-Bold/, "BoldItalic");
```

Comment:
Looking at `issue7835` it seems that we also need to handle `Arial-Black` as well, looking at PR #7839 we probably need to add a `replace("Arial-Black"`, "ArialBlack")` here as well?